gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/util/MetaFactoryUtil.java

```diff
@@ -0,0 +1,73 @@
+package uk.nhs.adaptors.pss.translator.util;
+
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Meta;
+import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.hl7.fhir.dstu3.model.UriType;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+public final class MetaFactoryUtil {
+
+    private MetaFactoryUtil() { }
+    private static final Coding NOPAT_CODING = new Coding()
+        .setSystem("http://hl7.org/fhir/v3/ActCode")
+        .setCode("NOPAT")
+        .setDisplay("no disclosure to patient, family or caregivers without attending provider's authorization");
+
+    public static Meta getMetaFor(final MetaType metaType, final String profile) {
+        return switch (metaType) {
+            case META_WITHOUT_SECURITY -> getBaseMeta(profile);
+            case META_WITH_SECURITY -> getMetaWithSecurity(profile);
+        };
+    }
+
+    public static Coding getNopatCoding() {
+        return NOPAT_CODING;
+    }
+
+    private static Meta getBaseMeta(final String profile) {
+        return new Meta().setProfile(
+            Collections.singletonList(
+                new UriType(profile)
+            )
+        );
+    }
+
+    private static Meta getMetaWithSecurity(final String profile) {
+        return getBaseMeta(profile).addSecurity(NOPAT_CODING);
+    }
+
+    public enum MetaType {
+        META_WITH_SECURITY,
+        META_WITHOUT_SECURITY
+    }
+
+    public static void assertMetaSecurityIsPresent(Meta initialMetaWithSecurityAdded, final Meta meta) {
+        final List<Coding> metaSecurity = meta.getSecurity();
+        final int metaSecuritySize = metaSecurity.size();
+
+        assertAll(
+            () -> assertThat(metaSecuritySize).isEqualTo(1),
+            () -> assertThat(meta).usingRecursiveComparison().isEqualTo(initialMetaWithSecurityAdded)
+        );
+    }
+
+    public static void assertMetaSecurityNotPresent(final ProcedureRequest request, String metaProfile) {
```

Comment:
Sliiightly strange to be coupled to a `ProcedureRequest`, I could see how you could pass in a generic `Resource` which might be more useful/reusable.

Reply:
thanks Adrian, a good point, yes, more generic with Resource is better here as it can be used by more Resource subtype classes. 👍